web/src/hooks/useMarketsPagination.ts

```diff
@@ -0,0 +1,21 @@
+import { Market } from "@/hooks/useMarket";
+import useMarketsSearchParams from "./useMarketsSearchParams";
+
+const itemsPerPage = 10;
+
+function useMarketsPagination(markets: Market[]) {
```

Comment:
this is paginating over the results initially returned by the graph (by default 100 markets), so when we have more than 100 markets some markets will not be displayed

we have two options:
1) increase the limit to 1000 to postpone this problem a bit further (1000 is the max allowed by the graph)
2) instead of doing the pagination on the client, paginate the query https://thegraph.com/docs/en/querying/graphql-api/#pagination (this allows to paginate more than 1000 markets)

(1) is ok for me, but if you want to do (2) the problem with paginating the subgraph query is that it's not possible to know the number of results to calculate how many pages the paginator needs to show. In this case, a "load more" type of pagination is better (react-query can handle it https://tanstack.com/query/v4/docs/framework/react/guides/paginated-queries , https://tanstack.com/query/latest/docs/framework/react/reference/useInfiniteQuery)

if possible I'd move this hook's logic to `useSortAndFilterMarkets`